doc/source/reference/routines.rst

```diff
@@ -23,7 +23,6 @@ indentation.
    routines.bitwise
    routines.char
    routines.strings
-   routines.ctypeslib
```

Comment:
Slightly confused why these are removed here.

Reply:
This listing is by design an opinionated way of splitting up NumPy's main functionality into separate topics. Its coverage is incomplete for what's in the main namespace, and many more niche/legacy/deprecated topics and namespaces are not covered. So these edits make the listed topics more sensible. It's only a first cut with the most obvious changes that (to me) are really no-no's to list here. It does require further improvements (example: there are two string topics, and the one with the more general name points to `numpy.char` not `numpy.strings`).

As for the specific entries removed here:
- `ctypeslib`: not relevant for 99.x% of users, so way too niche to list here - and now covered in the new module structure page
- `numpy.lib`: that's simply not a coherent topic at all, so an entry for it does not make sense
- `routines.matlib`: only contains a few functions and, more importantly, we're going to deprecate `matrix` usage and don't want any new code using it, so exposing this topic is not right anymore
- `testing.overrides`: way too niche to list here, folded into the regular `numpy.testing` page

Note that this is the second iteration of editing this section (after gh-25650), and there'll be at least one more. Probably after exercising the `.. legacy::` directive on a bunch of stuff.

Reply:
OK, that makes sense. Resolved for me, but leaving it open for possible other reviewers, just because it is hard to understand why this was done in isolation.

Reply:
I have now added a comment above this toctree for context, that should hopefully help keep things understandable and in better shape.